src/main/java/org/retest/rebazer/connector/GithubConnector.java

```diff
@@ -0,0 +1,136 @@
+package org.retest.rebazer.connector;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+import org.retest.rebazer.config.RebazerConfig.RepositoryConfig;
+import org.retest.rebazer.config.RebazerConfig.RepositoryTeam;
+import org.retest.rebazer.domain.PullRequest;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.web.client.RestTemplate;
+
+import com.jayway.jsonpath.DocumentContext;
+import com.jayway.jsonpath.JsonPath;
+
+public class GithubConnector implements RepositoryConnector {
+
+	private final static String baseUrl = "https://api.github.com/";
+
+	private final RestTemplate template;
+
+	public GithubConnector( final RepositoryTeam repoTeam, final RepositoryConfig repoConfig,
+			final RestTemplateBuilder builder ) {
+		final String basePath = "/repos/" + repoTeam.getName() + "/" + repoConfig.getName();
+
+		template = builder.basicAuthorization( repoTeam.getUser(), repoTeam.getPass() ).rootUri( baseUrl + basePath )
+				.build();
+	}
+
+	@Override
+	public PullRequest getLatestUpdate( final PullRequest pullRequest ) {
+		final DocumentContext jsonPath = jsonPathForPath( requestPath( pullRequest ) );
+		final PullRequest updatedPullRequest = PullRequest.builder() //
+				.id( pullRequest.getId() ) //
+				.source( pullRequest.getSource() ) //
+				.destination( pullRequest.getDestination() ) //
+				.lastUpdate( jsonPath.read( "$.updated_at" ) ) //
+				.build();
+		return updatedPullRequest;
+	}
+
+	@Override
+	public boolean isApproved( final PullRequest pullRequest ) {
+		final DocumentContext jsonPath = jsonPathForPath( requestPath( pullRequest ) + "/reviews" );
+		final List<String> states = jsonPath.read( "$..state" );
+		boolean approved = false;
+		for ( final String state : states ) {
+			if ( state.equals( "APPROVED" ) ) {
```

Comment:
Besides your lovely ```if ( a == true )``` (simplify) 😛, you are effectively checking the last state in the list. I assume you want to check if all or any are approved?
Streams help here a lot.

Reply:
Jep nice code, I have overlooked this in refactoring ^^